lightningd/lightningd.c

```diff
@@ -232,20 +370,37 @@ static void shutdown_subdaemons(struct lightningd *ld)
 		/* Removes itself from list as we free it */
 		tal_free(p);
 	}
+
+	/*~ Commit the transaction.  Note that the db is actually
+	 * single-threaded, so commits never fail and we don't need
+	 * spin-and-retry logic everywhere. */
 	db_commit_transaction(ld->wallet->db);
 }
 
+/*~ Chainparams are the parameters for eg. testnet vs mainnet.  This wrapper
+ * saves lots of struggles with our 80-column guideline! */
 const struct chainparams *get_chainparams(const struct lightningd *ld)
 {
+	/* "The lightningd is connected to the chain topology."
+	 * "The chain topology is connected to the bitcoind API."
+	 * "The bitcoind API is connected chain parameters."
+	 * -- Worst childhood song ever. */
 	return ld->topology->bitcoind->chainparams;
 }
 
+/*~ Our wallet logic needs to know what outputs we might be interested in: we
+ * keep the maximum-ever-used key index in the db, and add them all to the
+ * filter here. */
```

Comment:
maybe add a reference to BOLT 2 with the key derivation? For someone new to lightning it might not be clear yet that all keys are derived from a HD wallet

Reply:
Ack.